evalml/pipelines/components/transformers/samplers/base_sampler.py

```diff
@@ -170,20 +171,32 @@ def _initialize_oversampler(self, X, y, sampler_class):
             y (pd.Series): Target features
             sampler_class (imblearn.BaseSampler): The sampler we want to initialize
         """
-        _, y_ww = self._prepare_data(X, y)
+        _, y_pd = self._prepare_data(X, y)
         sampler_params = {
             k: v
             for k, v in copy.copy(self.parameters).items()
-            if k not in ["sampling_ratio", "sampling_ratio_dict"]
+            if k not in ["sampling_ratio", "sampling_ratio_dict", "k_neighbors_default"]
         }
         if self.parameters["sampling_ratio_dict"] is not None:
             # make the dictionary
-            dic = self._convert_dictionary(self.parameters["sampling_ratio_dict"], y_ww)
+            dic = self._convert_dictionary(self.parameters["sampling_ratio_dict"], y_pd)
         else:
             # create the sampling dictionary
             sampling_ratio = self.parameters["sampling_ratio"]
-            dic = make_balancing_dictionary(y_ww, sampling_ratio)
+            dic = make_balancing_dictionary(y_pd, sampling_ratio)
         sampler_params["sampling_strategy"] = dic
+
+        # check for k_neighbors value
+        neighbors = self.parameters["k_neighbors_default"]
+        min_counts = y_pd.value_counts().values[-1]
+        if min_counts <= 1:
+            raise ValueError(
+                f"Minority class needs more than 1 sample to use SMOTE!, received {min_counts} sample"
+            )
+        if min_counts <= neighbors:
+            neighbors = min_counts - 1
```

Comment:
I like the solution.  I just would prefer if the user knew what was happening.  Do we have a way that we can communicate to the user, who thinks that k_neighbors_default is one value that it's being reduced due to their input?

Reply:
I could introduce a `k_neighbors` parameter that becomes the value we set it to in the parameters dictionary? This way, users can see the parameter when they look at the param dic, but I don't think our other components currently log any changes they make internally. We don't raise any changes [here](https://github.com/alteryx/evalml/blob/0ec9206c3d8f3214404782b3c280bd1798562878/evalml/pipelines/components/estimators/classifiers/catboost_classifier.py#L64) and [here](https://github.com/alteryx/evalml/blob/0ec9206c3d8f3214404782b3c280bd1798562878/evalml/pipelines/components/estimators/classifiers/lightgbm_classifier.py#L77).

Does this sound fine to you @chukarsten?

Reply:
Uh, how about just a change to a docstring on the k_neighbors_default parameter?  I think that would be sufficient.